lib/iex/lib/iex.ex

```diff
@@ -40,7 +40,7 @@ defmodule IEx do
   the `ERL_AFLAGS` environment variable and make sure that it is set
   accordingly on your terminal/shell configuration.
 
-  On Linux:
+  On GNU/Linux:
```

Comment:
In that case it should be "Unix-like operating systems"
It could read something along the lines of:

> On Unix-like operating systems (GNU/Linux, Mac OS X, BSD, etc):

Reply:
Sounds great to me. :+1:

Reply:
Also linux systems can be built without GNU components, so just stating `GNU/Linux` is not correct either and the BEAM can still work as well, so just `Linux` is probably correct between those, however even Unix-like is not necessarily correct either as what is being shown is shell-specific, not Linux/BSD/Unix/Mac-like and someone can easily use other shells.  The line of code being shown is very Bash'y (Bash/ZSH/etc...) and won't work on shells like csh and others.

What should probably be done is to list the command as `Posix standard` and use the `env` command to set it instead, otherwise it should be listed as the `Bash-style shell` or so instead.  But it absolutely is not a Linux'ism (for example I don't use Bash).  And it is absolutely not Unix-like OS's either (as at work I *do* have the bash shell installed as a replacement for the command shell, not talking about the linux bash shell that Win10 has but the actual bash shell, and that command works on it on windows as well).

Reply:
I feel though that if you are not using bash, once you see a bash script, you will know the equivalent in whatever shell you are using.

Since this is documentation, I would prefer to make this the most accessible as possible rather than make this the most correct as possible and lose people along the way. Advanced users won't get lost.

Reply:
In that case I'd pick Unix-like or POSIX-like as those are popular and well defined terms then.

As for the Windows one, it really won't work on powershell, which is the default shell that comes up on Win10 at work (unsure if that's a group policy or is really default but...).

Reply:
Can you please send a PR that adds a "On Windows 10 / Powershell" section? Thank you :heart:

Reply:
Sent:  https://github.com/elixir-lang/elixir/pull/8511  :-)

Reply:
thank you for the write up of the situation @OvermindDL1.
So we should definitely center around the shell,
Dos you have the code that would be POSIX compliant?

Reply:
POSIX defines the hardware syscall interface, so just `set_env(var, value, flags)` and the programs built on them, which would be `env`, and being a program it cannot adjust the global environment settings of the 'existing' shell, rather it sets it in a new shell, thus the most generically compatible way to set an environment is to do `env KEY1=VAL1 KEY2=VAL2 ... <users-shell>` as that will add the environment variables to the `env` process than launch a new version of the existing shell with the environment that `env` currently has.  Although more general usage is to just run the command straight like `env BLAH=42 iex -s mix` or so.

However, in general there are a few main methods of setting a global environment value in the existing command line interpreter / shell, that is:

* Bash/Dash/ZSH/newer-KSH/etc: `export KEY=VAL`
* Bash/ZSH/all-KSH/etc: `KEY=VAL\nexport KEY`
* csh/tcsh/etc: `setenv KEY VAL`
* fish: `set -Ux KEY VAL`
* powershell: `$env:KEY = "VAL"`
* windows-cmd: `set KEY=VAL`

The `env KEY=VAL yourShell` will so ubiquitously work that if you have, say, github's tools installed on windows then `env BLAH=42 cmd` will even work on the windows command prompt, though env is not included in windows by default I don't think.

And there are more shells than that, but those are all the ones I've used.  My personal favorites are zsh and fish, though I tend to use zsh (with a ton of plugins) over fish due to it's bash compatibility though fish is by default a lot more user friendly.  :-)